managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -324,24 +326,25 @@ public Map<String, String> getCursorProperties() {
         return cursorProperties;
     }
 
-    @Override
-    public CompletableFuture<Void> setCursorProperties(Map<String, String> cursorProperties) {
+    private CompletableFuture<Void> asyncReadAndUpdateCursorProperties(
+            final Function<Map<String, String>, Map<String, String>> updateFunction) {
         CompletableFuture<Void> updateCursorPropertiesResult = new CompletableFuture<>();
         ledger.getStore().asyncGetCursorInfo(ledger.getName(), name, new MetaStoreCallback<>() {
```

Comment:
Can we remove the get operation?
Instead, we can only update the local properties cache if we get a BadVersion exception

Reply:
> Instead, we can only update the local properties cache if we get a BadVersion exception

Will this introduce inconsistency between broker memory state and zookeeper storage?